certified-connectors/DocuSignDemo/script.csx

```diff
@@ -1631,30 +1631,20 @@ public class Script : ScriptBase
       var query = HttpUtility.ParseQueryString(this.Context.Request.RequestUri.Query);
 
       var newSigner = new JObject();
+      newSigner["signer"] = null;
+      var newBody = new JObject();
       var recipientEmailId = query.Get("recipientEmail");
       var phoneNumber = query.Get("areaCode") + " " + query.Get("phoneNumber");
       var signerPhoneNumber = "";
       char[] charsToTrimPhoneNumber = { '*', ' ', '\'', '/', '-', '(', ')', '+'};
 
-      if ((recipientEmailId == null) && (query.Get("phoneNumber") == null))
-      {
-        throw new ConnectorException(HttpStatusCode.BadRequest, "ValidationFailure: Please fill either Recipient Email or Phone Number to retrieve Recipient information");
-      }
-
       if (recipientEmailId != null)
       {
         foreach(var signer in signers)
```

Comment:
consider rewriting to following to simply and for cleaner variable naming.
var signer = null;
for ( i =0 ; i < signers.Count; i++)
   {
      //match
      signer = signers[i]
   }

Reply:
The compiler is not letting me do this. 
error CS0815: Cannot assign <null> to an implicitly-typed variable
Apparently the type of the the variable "signer" need to be defined inorder to assign it to a ObjectArray. If I declare signer with the type as JObject( var signer1 = new JObject();), I see this exception: Cannot implicitly convert type 'Newtonsoft.Json.Linq.JToken' to 'Newtonsoft.Json.Linq.JObject'

Reply:
ok. try with `var signer = new JObject();` instead of assigning to null

Reply:
If I declare signer with the type as JObject( var signer = new JObject();), I see this exception: Cannot implicitly convert type 'Newtonsoft.Json.Linq.JToken' to 'Newtonsoft.Json.Linq.JObject'

Reply:
signers[i] is a JToken and not JObject

Reply:
Done